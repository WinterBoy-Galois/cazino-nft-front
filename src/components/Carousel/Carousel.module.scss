@import '../../styles/shared';

.container {
  margin: auto;
  width: 100%;
  overflow: auto;

  @include media-breakpoint-up(md) {
    width: 400px;
  }

  @include media-breakpoint-up(lg) {
    width: 530px;
  }

  @include media-breakpoint-up(xl) {
    width: 750px;
  }
}

.slide {
  height: 82px;
  border-radius: 15px;
  line-height: 1;
  background-color: #182839;
  background-repeat: no-repeat;
  background-size: auto;
  background-position: right top;
  background-image: url('../../assets/images/carousel/carousel1-home.svg');
  overflow: hidden;
  user-select: none;

  @include media-breakpoint-up(xl) {
    height: 180px;
  }
}

.swiper {
  &__pagination {
    bottom: 95%;
    height: 16px;
    width: 16px;
    background-color: #32506a;
    border: 1px solid #132739;
    border-radius: 16px;
    margin-left: 16px;
    outline: none;
    opacity: 0.6;
    transition: opacity 0.3s ease;
    padding: 0;
  }

  .swiper-pagination {
    color: #fff;
  }
}
.carousel {
  &__slide-inner {
    margin: auto 0;
    width: 50%;
    padding: 10px;

    @include media-breakpoint-up(xl) {
      padding: 30px;
      width: 60%;
    }
  }

  &__dot-group {
    text-align: center;
    margin-top: 8px;
    margin-bottom: 8px;

    @include media-breakpoint-up(xl) {
      margin-top: 21px;
      margin-bottom: 21px;
    }

    button {
      height: 16px;
      width: 16px;
      background-color: #32506a;
      border: 1px solid #132739;
      border-radius: 16px;
      margin-left: 16px;
      outline: none;
      opacity: 0.6;
      transition: opacity 0.3s ease;
      padding: 0;

      @include media-breakpoint-up(xl) {
        margin-top: 0;
        height: 21px;
        width: 21px;
        border-radius: 21px;
        margin-left: 21px;
      }

      &:hover {
        opacity: 1;
      }

      &:disabled {
        cursor: default;
        background-color: #132739;
        border: 1px solid #32506a;
      }
    }
  }
}

.headline {
  color: #ffd800;
  font-size: 14px;
  margin: 0;
  font-weight: 300;
  margin-bottom: 0.1rem;
  margin-top: 0.8rem;

  @include media-breakpoint-up(md) {
    margin-top: 0;
  }

  @include media-breakpoint-up(xl) {
    font-size: 28px;
    margin-bottom: 0.4rem;
    margin-top: 0;
  }
}

.subline {
  color: #4d566c;
  font-size: 14px;
  font-weight: 300;
  margin-top: 14px;
  display: none;

  @include media-breakpoint-up(md) {
    display: block;
    margin-top: 7px;
  }

  @include media-breakpoint-up(xl) {
    font-size: 20px;
    margin-top: 20px;
    display: block;
  }
}
