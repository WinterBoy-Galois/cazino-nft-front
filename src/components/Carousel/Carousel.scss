@import '../../styles/shared';
@import 'swiper/swiper.scss';
@import 'swiper/components/pagination/pagination.scss';

.swiper-wrapper {
  padding-bottom: 42px;

  @include media-breakpoint-up(lg) {
    padding-bottom: 55px;
  }
}

.swiper-pagination {
  text-align: center;

  &-bullet {
    height: 16px;
    width: 16px;
    background-color: #32506a;
    border: 1px solid #132739;
    border-radius: 16px;
    margin-left: 16px;
    outline: none;
    opacity: 0.6;
    transition: opacity 0.3s ease;
    padding: 0;
    display: inline-block;

    @include media-breakpoint-up(lg) {
      height: 21px;
      width: 21px;
      border-radius: 21px;
      margin-left: 21px;
    }

    &-active {
      cursor: default;
      background-color: #132739;
      border: 1px solid #32506a;
    }

    &:hover {
      opacity: 1;
    }
  }
}

.swiper-container-horizontal {
  > .swiper-pagination-bullets {
    span:first-of-type {
      margin-left: 0;
    }
    .swiper-pagination-bullet {
      margin: 0 0 0 16px;
    }
  }
}
