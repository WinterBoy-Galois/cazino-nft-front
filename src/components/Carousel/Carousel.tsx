import React, { useState, useEffect } from 'react';
import { Swiper, SwiperSlide } from 'swiper/react';
import SwiperCore, { Autoplay, Pagination } from 'swiper';
import styles from './Carousel.module.scss';
import { useStateValue } from '../../state';

const Carousel: React.FC = () => {
  const [swiper, setSwiper] = useState<SwiperCore | null>();

  const [
    {
      sidebar: { isOpen },
    },
  ] = useStateValue();

  useEffect(() => {
    setTimeout(() => {
      swiper?.update();
    }, 220);
  }, [isOpen, swiper]);

  const params = {
    spaceBetween: 16,
    speed: 600,
    pagination: {
      // el: `.swiper-pagination`,
      clickable: true,
    },
    loop: true,
    autoplay: {
      delay: 5000,
      disableOnInteraction: true,
    },
  };

  SwiperCore.use([Autoplay, Pagination]);

  return (
    <div className={styles.container}>
      <Swiper {...params} onSwiper={swiper => setSwiper(swiper)}>
        <SwiperSlide className={styles.slide}>
          <div className={styles['carousel__slide-inner']}>
            <div className={styles.headline}>
              Your friends make <strong>BETS</strong>
              <br /> you get <strong>BITS</strong>!
            </div>
            <div className={styles.subline}>
              Best referral program comissions you will ever find
            </div>
          </div>
        </SwiperSlide>
        <SwiperSlide className={styles.slide}>
          <div className={styles['carousel__slide-inner']}>
            <div className={styles.headline}>
              I&apos;m the <strong>SECOND</strong> Slide.
            </div>
          </div>
        </SwiperSlide>
        <SwiperSlide className={styles.slide}>
          <div className={styles['carousel__slide-inner']}>
            <div className={styles.headline}>
              I&apos;m the <strong>THIRD</strong> Slide.
            </div>
          </div>
        </SwiperSlide>
        <SwiperSlide className={styles.slide}>
          <div className={styles['carousel__slide-inner']}>
            <div className={styles.headline}>
              I&apos;m the <strong>FOURTH</strong> Slide.
            </div>
          </div>
        </SwiperSlide>
        <SwiperSlide className={styles.slide}>
          <div className={styles['carousel__slide-inner']}>
            <div className={styles.headline}>
              I&apos;m the <strong>FIFTH</strong> Slide.
            </div>
          </div>
        </SwiperSlide>
      </Swiper>
    </div>
  );
};

export default Carousel;
