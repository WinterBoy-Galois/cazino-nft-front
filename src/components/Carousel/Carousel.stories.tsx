import React from 'react';
import Layout from '../Layout';
import Carousel from '.';
import GameButtonList from '../GameButtonList';
import { LocationProvider } from '@reach/router';
import { MockedProvider } from '@apollo/client/testing';
import { BET_ADDED } from '../../graphql/subscriptions';

const mocks = [
  {
    request: {
      query: BET_ADDED,
    },
    result: {
      data: {},
    },
  },
];

export default {
  title: 'Components/Carousel',
  component: Carousel,
  decorators: [
    (storyFn: () => React.ReactElement) => (
      <MockedProvider mocks={mocks} addTypename={false}>
        {storyFn()}
      </MockedProvider>
    ),
  ],
};

export const Default = () => <Carousel />;

export const WithLayout = () => (
  <LocationProvider>
    <Layout>
      <div className="container">
        <Carousel />
      </div>
      <div className="container">
        <GameButtonList />
      </div>
    </Layout>
  </LocationProvider>
);
