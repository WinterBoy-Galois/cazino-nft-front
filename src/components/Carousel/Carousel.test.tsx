import React from 'react';
import { render } from '@testing-library/react';
import Carousel from './Carousel';

describe('Carousel', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<Carousel />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
