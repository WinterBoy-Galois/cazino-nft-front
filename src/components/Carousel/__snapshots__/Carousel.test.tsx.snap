// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="swiper-container swiper-container-initialized swiper-container-horizontal"
        >
          <div
            class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets"
          />
          <div
            class="swiper-wrapper"
            style="transition: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
          >
            <div
              class="swiper-slide swiper-slide-duplicate slide"
              data-swiper-slide-index="4"
            >
              <div
                class="carousel__slide-inner"
              >
                <div
                  class="headline"
                >
                  I'm the 
                  <strong>
                    FIFTH
                  </strong>
                   Slide.
                </div>
              </div>
            </div>
            <div
              class="swiper-slide slide"
              data-swiper-slide-index="0"
            >
              <div
                class="carousel__slide-inner"
              >
                <div
                  class="headline"
                >
                  Your friends make 
                  <strong>
                    BETS
                  </strong>
                  <br />
                   you get 
                  <strong>
                    BITS
                  </strong>
                  !
                </div>
                <div
                  class="subline"
                >
                  Best referral program comissions you will ever find
                </div>
              </div>
            </div>
            <div
              class="swiper-slide slide"
              data-swiper-slide-index="1"
            >
              <div
                class="carousel__slide-inner"
              >
                <div
                  class="headline"
                >
                  I'm the 
                  <strong>
                    SECOND
                  </strong>
                   Slide.
                </div>
              </div>
            </div>
            <div
              class="swiper-slide slide"
              data-swiper-slide-index="2"
            >
              <div
                class="carousel__slide-inner"
              >
                <div
                  class="headline"
                >
                  I'm the 
                  <strong>
                    THIRD
                  </strong>
                   Slide.
                </div>
              </div>
            </div>
            <div
              class="swiper-slide slide"
              data-swiper-slide-index="3"
            >
              <div
                class="carousel__slide-inner"
              >
                <div
                  class="headline"
                >
                  I'm the 
                  <strong>
                    FOURTH
                  </strong>
                   Slide.
                </div>
              </div>
            </div>
            <div
              class="swiper-slide slide"
              data-swiper-slide-index="4"
            >
              <div
                class="carousel__slide-inner"
              >
                <div
                  class="headline"
                >
                  I'm the 
                  <strong>
                    FIFTH
                  </strong>
                   Slide.
                </div>
              </div>
            </div>
            <div
              class="swiper-slide swiper-slide-duplicate slide"
              data-swiper-slide-index="0"
            >
              <div
                class="carousel__slide-inner"
              >
                <div
                  class="headline"
                >
                  Your friends make 
                  <strong>
                    BETS
                  </strong>
                  <br />
                   you get 
                  <strong>
                    BITS
                  </strong>
                  !
                </div>
                <div
                  class="subline"
                >
                  Best referral program comissions you will ever find
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="swiper-container swiper-container-initialized swiper-container-horizontal"
      >
        <div
          class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets"
        />
        <div
          class="swiper-wrapper"
          style="transition: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide swiper-slide-duplicate slide"
            data-swiper-slide-index="4"
          >
            <div
              class="carousel__slide-inner"
            >
              <div
                class="headline"
              >
                I'm the 
                <strong>
                  FIFTH
                </strong>
                 Slide.
              </div>
            </div>
          </div>
          <div
            class="swiper-slide slide"
            data-swiper-slide-index="0"
          >
            <div
              class="carousel__slide-inner"
            >
              <div
                class="headline"
              >
                Your friends make 
                <strong>
                  BETS
                </strong>
                <br />
                 you get 
                <strong>
                  BITS
                </strong>
                !
              </div>
              <div
                class="subline"
              >
                Best referral program comissions you will ever find
              </div>
            </div>
          </div>
          <div
            class="swiper-slide slide"
            data-swiper-slide-index="1"
          >
            <div
              class="carousel__slide-inner"
            >
              <div
                class="headline"
              >
                I'm the 
                <strong>
                  SECOND
                </strong>
                 Slide.
              </div>
            </div>
          </div>
          <div
            class="swiper-slide slide"
            data-swiper-slide-index="2"
          >
            <div
              class="carousel__slide-inner"
            >
              <div
                class="headline"
              >
                I'm the 
                <strong>
                  THIRD
                </strong>
                 Slide.
              </div>
            </div>
          </div>
          <div
            class="swiper-slide slide"
            data-swiper-slide-index="3"
          >
            <div
              class="carousel__slide-inner"
            >
              <div
                class="headline"
              >
                I'm the 
                <strong>
                  FOURTH
                </strong>
                 Slide.
              </div>
            </div>
          </div>
          <div
            class="swiper-slide slide"
            data-swiper-slide-index="4"
          >
            <div
              class="carousel__slide-inner"
            >
              <div
                class="headline"
              >
                I'm the 
                <strong>
                  FIFTH
                </strong>
                 Slide.
              </div>
            </div>
          </div>
          <div
            class="swiper-slide swiper-slide-duplicate slide"
            data-swiper-slide-index="0"
          >
            <div
              class="carousel__slide-inner"
            >
              <div
                class="headline"
              >
                Your friends make 
                <strong>
                  BETS
                </strong>
                <br />
                 you get 
                <strong>
                  BITS
                </strong>
                !
              </div>
              <div
                class="subline"
              >
                Best referral program comissions you will ever find
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
